### ISSUE TITLE
fs.removeSync silently fails

### ISSUE THREAD

**Issue by FredrikJohansson83** [jprichardson/node-fs-extra#407]
When using `fs.removeSync` in a `try/catch` block, it silently fails. However, if I instead use `fs.remove` with an error callback function, it does end up in the callback method.

```
try {
    fs.removeSync('C:\\Projects\\testdir');
} catch (err) { 
    console.log("Error", err); 
}
```
> Gives no output, but directory is not removed when looking in Windows.

```
fs.removeSync('C:\\Projects\\testdir', function(err) { 
    console.log("Error", err); 
});
```
> Gives me the following output:
```
{ Error: EBUSY: resource busy or locked, rmdir 'C:\Projects\testdir'
    at Error (native)
  errno: -4082,
  code: 'EBUSY',
  syscall: 'rmdir',
  path: 'C:\\Projects\\testdir' }
```

Using the following:
> Windows 10
> node v6.10.1
> fs-extra v2.1.2

**Comment by RyanZim** [comment]
Ugh, looks like another windows bug. Issues like this are hard to track down since I can't actually test them locally.

I'm looking at https://github.com/jprichardson/node-fs-extra/blob/master/lib/remove/rimraf.js#L277; there's no `else` statement there; I wonder if that's the issue.

**Comment by RyanZim** [comment]
@FredrikJohansson83 Give this branch a try: https://github.com/jprichardson/node-fs-extra/tree/rmSync-patch

Run:
```
npm uninstall node-fs-extra
npm install jprichardson/node-fs-extra#rmSync-patch
```
Then test again. Let me know your results.

**Comment by FredrikJohansson83** [comment]
@RyanZim Thanks! I tried it and it seems to work! One difference compared to the async error callback though, it seems to include a stack trace of the error, but maybe this is how it's supposed to work when using it in a try/catch?

```
try { 
    fs.removeSync('C:\\Projects\\testdir') 
} catch (err) { 
    console.log("err", err); 
}
```

Gives the following output:
```
err { Error: EBUSY: resource busy or locked, rmdir 'C:\Projects\testdir'
    at Error (native)
    at Object.fs.rmdirSync (fs.js:887:18)
    at rmdirSync (C:\Projects\git\...\node_modules\fs-extra\lib\remove\rimraf.js:276:13)
    at Function.rimrafSync [as sync] (C:\Projects\git\...\node_modules\fs-extra\lib\remove\rimraf.js:252:7)
    at Object.removeSync (C:\Projects\git\...\node_modules\fs-extra\lib\remove\index.js:6:17)
    at repl:1:9
    at sigintHandlersWrap (vm.js:22:35)
    at sigintHandlersWrap (vm.js:73:12)
    at ContextifyScript.Script.runInThisContext (vm.js:21:12)
    at REPLServer.defaultEval (repl.js:340:29)
  errno: -4082,
  code: 'EBUSY',
  syscall: 'rmdir',
  path: 'C:\\Projects\\testdir' }
```

**Comment by RyanZim** [comment]
I think that looks like correct behavior. PR coming!

**Comment by RyanZim** [comment]
PR done: #408

**Comment by FredrikJohansson83** [comment]
@RyanZim Great! Thanks for your support!

**Comment by Clonkex** [comment]
For anyone else encountering this issue, it appears that fs-extra switched to using Node's own `fs.rmSync` method at some point, and `fs.rmSync` also had [this issue](https://github.com/nodejs/node/issues/38683) at least as recently as 16.15.0. I've tested and confirmed it's fixed in 18.12.0, but probably also in 17.x.
